dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt

```diff
@@ -38,6 +38,7 @@ internal class RumSessionScope(
     internal val activeChildrenScopes = mutableListOf<RumScope>()
 
     internal var keepSession: Boolean = false
+    @Volatile
```

Comment:
why do we need this? I can see it is updated from 2 places: `handleEvent` (in case of `ResetSession` event coming) and from `updateSessionIdIfNeeded`. But latter is `Synchronized`, so field should have good state after the call, so it leaves us only with `handleEvent(ResetSession)` case, but in this case shouldn't `resetSessionTime` and `applicationDisplayed` be volatile as well (since they updated also during this event)?

Reply:
`applicationDisplayed` is volatile by default ( it is a primitive) . Normally there shouldn't be any problem of visibility as this is updated only from "Executors.newSingleThreadExecutor()" which normally should use only one single thread but just wanted to exclude this possibility in any case.

Reply:
Hmm, I am taking back what I said about that not primitives being volatile by default. It seems they're actually not.

Reply:
I am going to remove the `@Volatile` annotation on that reference as I checked the `SingleThreadExecutor` and it will always keep the thread alive (not recycling it) so we will not get into thread memory visibility issues. But just to keep in mind that primitives are not volatile by default in JVM.